### DISCUSSION TITLE
Change precision of codegen datatype

### DISCUSSION THREAD

**Discussion by matthiasfrontull** [casadi/casadi#3524] [Q&A]
Hello! 
I want to deploy c code generated by casadi on a µC with single precision. 
My approach would be to change somehow the default datatype of the casadi_real from double to single. 
It that possible or is there a better solution to that problem?

Thanks in advance
Matthews

**Comment by jgillis** [comment]
The 'somehow' can be done in two ways:

1. When code-generating, you can pass an options dictionary. One of the allowed entries is `casadi_real` which you can set to any string of your choice.
2. After code-generating, you can compile with flags that `#define` `casadi_real` to be something custom.

**Comment by matthiasfrontull** [comment]
Hello Mr. Gillis,
Thanks a lot for the quick response and the hints.
Best regardsMatthias

Gesendet von Yahoo Mail auf Android 
 
  Am Di., Jan. 9, 2024 at 15:23 schrieb Joris ***@***.***>:   


The 'somehow' can be done in two ways:
   
   - When code-generating, you can pass an options dictionary. One of the allowed entries is casadi_real which you can set to any string of your choice.
   - After code-generating, you can compile with flags that #define casadi_real to be something custom.
